torch/csrc/jit/script/compiler.cpp

```diff
@@ -111,6 +111,23 @@ struct Environment {
 
     return sv;
   }
+
+  SugaredValuePtr createCapturedInputIfNeeded(const SourceRange& loc, std::string ident) {
+    auto in_frame = findInThisFrame(ident);
+    if (in_frame)
+      return in_frame;
+
+    // recursively handles the case where parent blocks are also loops
+    auto from_parent = next ? next->createCapturedInputIfNeeded(loc, ident) : nullptr;
+
+    // recursively create the captured input if it is the loop block
+    Value* simple_val = asSimple(from_parent);
```

Comment:
This looks pretty worrying. What if `from_parent` is a `nullptr`?

Reply:
Good catch! updated to avoid nullptr problem, thanks!